### PR TITLE
[Application Runtime] Pass body when invoke api gateway

### PR DESCRIPTION
**Pull request by rokatyy**
Jira -  https://iguazio.atlassian.net/browse/ML-7573

### CODE REVIEW

File: mlrun/runtimes/nuclio/application/application.py [hunk 1]
```diff
@@ -509,6 +509,13 @@ def invoke(
 
         if not method:
             method = "POST" if body else "GET"
+
+        # Determine the correct keyword argument for the body
+        if isinstance(body, dict):
+            http_client_kwargs["json"] = body
+        elif isinstance(body, (str, bytes)):
+            http_client_kwargs["data"] = body
```
Comment (TomerShor) [inline mlrun/runtimes/nuclio/application/application.py]: We should add the `body` argument to the api_gateway's `invoke` function, and this should be moved there.
Then body will be supported both in application and api gateway in general

**Review by TomerShor — CHANGES_REQUESTED** [review]
Looks fine but have a suggestion

File: mlrun/runtimes/nuclio/application/application.py (hunk 1 above)
Comment (rokatyy) [reply mlrun/runtimes/nuclio/application/application.py]: @TomerShor api gw already supports data/json - whatever user sets in kwargs, it's just a matter of decision, but I think both options are fine, so will move to gws

File: mlrun/runtimes/nuclio/api_gateway.py [hunk 2]
```diff
@@ -386,6 +386,7 @@ def invoke(
         headers: dict = None,
         credentials: Optional[tuple[str, str]] = None,
         path: Optional[str] = None,
+        body: typing.Union[str, bytes, dict] = None,
```
Comment (TomerShor) [inline mlrun/runtimes/nuclio/api_gateway.py]:
```suggestion
        body: Optional[typing.Union[str, bytes, dict]] = None,
```

File: mlrun/runtimes/nuclio/application/application.py [hunk 3]
```diff
@@ -509,6 +509,7 @@ def invoke(
 
         if not method:
             method = "POST" if body else "GET"
+
         return self.status.api_gateway.invoke(
```
Comment (TomerShor) [inline mlrun/runtimes/nuclio/application/application.py:513]: Add `body=body` to the invoke call

**Review by TomerShor — CHANGES_REQUESTED** [review]

File: mlrun/runtimes/nuclio/application/application.py (hunk 3 above)
Comment (rokatyy) [reply mlrun/runtimes/nuclio/application/application.py:513]: right, nice catch

**Review by moranbental — APPROVED** [review]
LGTM

**Review by TomerShor — APPROVED** [review]

**Pull request merged**